src/extension.node.ts

```diff
@@ -3,13 +3,8 @@
 
 'use strict';
 
-/* eslint-disable @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports */
-
-// This line should always be right on top.
-/* eslint-disable @typescript-eslint/no-explicit-any */
-if ((Reflect as any).metadata === undefined) {
-    require('reflect-metadata');
-}
+// reflect-metadata is needed by inversify, this must come before any inversify references
+import 'reflect-metadata';
```

Comment:
We cannot do this, its possible that another extension also uses `reflect-metadata` as a result of importing this again the module will blow away all of the previous data (causing other extension that use `reflect-metadta` to fail.

Hence we need to check if the global variable already exists.

Reply:
I.e. it must be conditionally imported

Reply:
TIL https://github.com/inversify/InversifyJS/issues/737